### PR TITLE
flutter_quill_extensions: replace gallery_saver with gal

### PR DESCRIPTION
**Pull request by santiagos01**
Resolves https://github.com/singerdmx/flutter-quill/issues/1377

### CODE REVIEW

**Comment by singerdmx** [comment]
@Jon-Salmon could you please publish a new version of package?

File: flutter_quill_extensions/lib/embeds/builders.dart [hunk 1]
```diff
@@ -266,13 +269,31 @@ Widget _menuOptionsForReadonlyImage(
...
+                  if (response.statusCode != 200) {
+                    throw Exception(
+                      'failed to download image: ${response.statusCode}',
+                    );
+                  }
+
+                  // Save image to a temporary path
+                  final fileName = uri.pathSegments.isEmpty ? 'image.jpg'
+                      : uri.pathSegments.last;
+                  final imagePath = '${Directory.systemTemp.path}/menu-opt-$fileName';
```
Comment (Jon-Salmon) [inline flutter_quill_extensions/lib/embeds/builders.dart:287]: This should use a cache directory from path_provider. This method here may not work on android:
https://github.com/flutter/flutter/issues/12116
https://github.com/NativeScript/android/issues/828

File: flutter_quill_extensions/lib/embeds/builders.dart [hunk 2]
```diff
@@ -266,13 +269,31 @@ Widget _menuOptionsForReadonlyImage(
...
+                  final imageFile = File(imagePath);
+                  await imageFile.writeAsBytes(response.bodyBytes);
+
+                  // Save image to gallery
+                  await Gal.putImage(imagePath);
```
Comment (Jon-Salmon) [inline flutter_quill_extensions/lib/embeds/builders.dart:292]: Should probably delete the temporary image after saving

File: flutter_quill_extensions/pubspec.yaml [hunk 3]
```diff
@@ -14,11 +14,12 @@ dependencies:
 
   flutter_quill: ^7.2.19
 
+  http: ^1.1.0
   image_picker: ">=0.8.5 <2.0.0"
   photo_view: ^0.14.0
   video_player: ^2.7.0
   youtube_player_flutter: ^8.1.1
-  gallery_saver: ^2.3.2
+  gal: ^2.1.1
```
Comment (Jon-Salmon) [inline flutter_quill_extensions/pubspec.yaml:22]: The `gal` package requires some setup steps that the user has to take on install. These should be linked to from the readme

**Review by Jon-Salmon — COMMENTED** [review]
@santiagos01 There are some review items that need to be addressed before this can be released

**Comment by LuchoTurtle** [comment]
@Jon-Salmon , given the feedback you've given, are you working on a PR to tackle these issues? If not, I could try and give it a whirl and open it myself. It would really be awesome if we could get a new package version going with these changes. The `http` dependency versioning error seems to be a put-off for a lot of people, so this would work wonderfuly! :D

**Comment by Jon-Salmon** [comment]
@LuchoTurtle Please feel free to as I don't have time to work on this currently. I was hoping @santiagos01 would update this but I've not heard anything yet

File: flutter_quill_extensions/lib/embeds/builders.dart (hunk 1 above)
Comment (santiagos01) [reply flutter_quill_extensions/lib/embeds/builders.dart:287]: Oops, sorry, didn't notice that this doesn't work on android.
Has this been resolved?

File: flutter_quill_extensions/lib/embeds/builders.dart (hunk 2 above)
Comment (santiagos01) [reply flutter_quill_extensions/lib/embeds/builders.dart:292]: Ah right.., has this been resolved?

**Comment by santiagos01** [comment]
Hi all, sorry just saw this again --

Can I check whether the issues mentioned above have been resolved?

**Comment by LuchoTurtle** [comment]
I believe the issues have already been fixed through https://github.com/singerdmx/flutter-quill/pull/1414, @santiagos01

File: flutter_quill_extensions/lib/embeds/builders.dart (hunk 1 above)
Comment (LuchoTurtle) [reply flutter_quill_extensions/lib/embeds/builders.dart:287]: I think https://github.com/singerdmx/flutter-quill/pull/1414 already fixed these.

**Pull request merged**